### ISSUE TITLE
cloudstack_network aclid to acl_id migration forces new resource creation 

### ISSUE THREAD

**Issue by landro** [hashicorp/terraform#6630]
### Terraform Version
0.6.15

### Affected Resource(s)
cloudstack_network

### Terraform Configuration Files

From 

```hcl
resource "cloudstack_network" "log" {
  name             = "Log network"
  vpc_id           = "${cloudstack_vpc.default.id}"
  cidr             = "${lookup(var.cidr, "log")}"
  network_offering = "${var.default_networkoffering}"
  zone             = "${var.default_zone}"
  aclid            = "${cloudstack_network_acl.log.id}"
}
```

To 

```hcl
resource "cloudstack_network" "log" {
  name             = "Log network"
  vpc_id           = "${cloudstack_vpc.default.id}"
  cidr             = "${lookup(var.cidr, "log")}"
  network_offering = "${var.default_networkoffering}"
  zone             = "${var.default_zone}"
  acl_id            = "${cloudstack_network_acl.log.id}"
}
```

### Expected Behavior
No changes

### Actual Behavior
-/+ cloudstack_network.log
    acl_id:           "" => "6dead3a3-91d1-4418-9af1-6ccf30c77a33" (forces new resource)
    aclid:            "6dead3a3-91d1-4418-9af1-6ccf30c77a33" => ""

**Comment by svanharmelen** [comment]
Thanks for the report @landro! I must admit this tents to look like a bug, but it's a little more complicated than that :wink:

The problem is that we need to set `ForceNew` on this attribute to make sure that if you change it, it will take the correct action. But now when you change your plan to use the updated name, TF core (correctly) also sees that as a change and so I will plan to recreate the resource.

If I'm right, first doing a `terraform refresh` should solve this issue as that will read back the ACL ID used by this network into the `acl_id` param. So doing a plan after that, so not see this as a diff...

Did you by any change run the plan job by `terraform plan -refresh=false`? As I would suspect this should also fix it even without running the `refresh ` command.

Please let me know if my assumptions are correct and this fixes the behaviour, because if not I'll have to dive into the code and do some more testing.

**Comment by landro** [comment]
@svanharmelen - the output provided above actually came from running  _terraform plan_ without the _-refresh=false_

**Comment by svanharmelen** [comment]
Strange... And the explicit `terraform refresh`, does that help?

**Comment by svanharmelen** [comment]
@landro I know how I can fix this, will add a PR in a few...

**Comment by svanharmelen** [comment]
This is solved in PR #6741

**Comment by landro** [comment]
Great new @svanharmelen ! Thanks for the effort!
